great_expectations/rule_based_profiler/domain_builder/domain_builder.py

```diff
@@ -55,6 +55,19 @@ def get_domains(
         """
         return self._get_domains(variables=variables)
 
+    @property
+    @abstractmethod
+    def domain_type(self) -> Union[str, MetricDomainTypes]:
+        pass
```

Comment:
Same note here - why is it both str and MetricDomainType? Users can use strings in lieu of the enum but the underlying type is indeed that `MetricDomainTypes` right?

Reply:
@cdkini Because we allow users to use the string version in kwargs (and then we validate it).

Reply:
I think this should just the enum type - the user provides a string but the `__eq__` works between the string version and the enum. The ultimate type should be the enum here.

Reply:
@cdkini We get type errors then...